common/types/nodeid.go

```diff
@@ -9,6 +9,18 @@ import (
 	"github.com/spacemeshos/go-spacemesh/log"
 )
 
+var ownNodeID NodeID
```

Comment:
i don't think that adding global public key is acceptable even as a mitigation.

beside that this code already needs to be removed next week